capplets/appearance/appearance.h

```diff
@@ -137,3 +137,4 @@ typedef struct {
 } AppearanceData;
 
 #define appearance_capplet_get_widget(x, y) (GtkWidget*) gtk_builder_get_object(x->ui, y)
+#define GET_WINDOW(x) (GTK_WINDOW (gtk_builder_get_object (data->ui, (x))))
```

Comment:
I don't like macros that use local variables implicitly, unless they are highly specific and defined right next to the code using them.  Here, I'd really prefer seeing `data` be a parameter -- which basically makes it `GTK_WINDOW (appearance_capplet_get_widget (x, y))`.

The biggest reason why I think it's better is that implicit use of local variables make it a lot harder to understand when reading, and a lot easier to debug would anything go wrong (e.g. a `undefined variable 'data'` would be cryptic for the caller, so would `'data' does not have a member named 'ui'` or such).  It also makes it slightly harder to use, as if you happen to already have a variable `data` you need to refactor your code to be able to use that macro.

Reply:
I don't think it's necessary to add a new variable in order to predict a future refactoring, such as builder on the sample below:

```
GtkBuilder *builder = data->ui;
...
... GET_WINDOW (label) ...
...
#define GET_WINDOW(x) (GTK_WINDOW (gtk_builder_get_object (builder, (x))))
```

Reply:
I don't think we need a new magic variable, I think we need an additional parameter to the `GET_WINDOW()` macro:
```c
#define GET_WINDOW(appearance_data, y) (GTK_WINDOW (appearance_capplet_get_widget (appearance_data, y)))
```
and call it as `GET_WINDOW (data, "some member")`.

Reply:
You do not need to add a new parameter, if it always takes the same value.

Reply:
My point is that it adds magic to that macro, and requirements on the environment in which it is called.  Both of these make it harder to use, and prone to errors or headaches in the future.  And I don't think saving typing 6 characters (`data, `) is worth this kind of trouble.
In other words, I don't think having one parameter less is not worth requiring the caller to have a variable with a specific name and semantic defined to be able to use this macro.

However, feel free to ignore my remarks, but basically my experience is "magic" in macros leads sooner or later to problems, unless it is *very* clear and well understood by all parties (callers and callee).

Reply:
In the case you describe you would no longer need to use the GET_WINDOW macro and you would use the value replaced by the processor directly.

Reply:
Well, using the macro would save the caller from some typing, which is what I though your goal was.  Instead of `(GTK_WINDOW (appearance_capplet_get_widget (data, "something")))`, the caller would type `GET_WINDOW (data, "something")`.